modules/convert/convert.go

```diff
@@ -414,3 +415,32 @@ func ToLFSLock(l *git_model.LFSLock) *api.LFSLock {
 		},
 	}
 }
+
+// ToChangedFile convert a gitdiff.DiffFile to api.ChangedFile
+func ToChangedFile(f *gitdiff.DiffFile, repo *repo_model.Repository, commit string) *api.ChangedFile {
+	status := "changed"
+	if f.IsDeleted {
+		status = "deleted"
+	} else if f.IsCreated {
+		status = "added"
+	} else if f.IsRenamed && f.Type == gitdiff.DiffFileCopy {
+		status = "copied"
+	} else if f.IsRenamed && f.Type == gitdiff.DiffFileRename {
+		status = "renamed"
+	}
+
+	file := &api.ChangedFile{
+		Filename:    f.GetDiffFileName(),
+		Status:      status,
+		Additions:   f.Addition,
+		Deletions:   f.Deletion,
+		HTMLURL:     fmt.Sprint(repo.HTMLURL(), "/src/commit/", commit, "/", util.PathEscapeSegments(f.GetDiffFileName())),
+		ContentsURL: fmt.Sprint(repo.HTMLURL(), "/raw/commit/", commit, "/", util.PathEscapeSegments(f.GetDiffFileName())),
```

Comment:
This should be `RawURL`

Reply:
Hmm, if we want to follow GitHub, then there needs more changes:
* GitHub's `blob_url` is our current `html_url`, although in the GitHub contents API the URL is called as `html_url` again
* GitHub's `raw_url` is our current `contents_url` (as above)
* GitHub's `contents_url` is our API url (as above)

Just paste my some test PRs here from another comment (for reference)

Test PR1:  https://github.com/wxiaoguang/playground/pull/1 , https://api.github.com/repos/wxiaoguang/playground/pulls/1/files

Test PR2:  https://github.com/wxiaoguang/playground/pull/2 , https://api.github.com/repos/wxiaoguang/playground/pulls/2/files